ouroboros-consensus/src/Ouroboros/Consensus/Fragment/InFuture.hs

```diff
@@ -115,15 +112,9 @@ reference :: forall m blk. (Monad m, UpdateLedger blk, HasHardForkHistory blk)
           -> SystemTime m
           -> CheckInFuture m blk
 reference cfg (ClockSkew clockSkew) SystemTime{..} = CheckInFuture {
-      checkInFuture = \validated -> do
-        now <- systemTimeCurrent
-        -- Since we have the ledger state /after/ the fragment, the derived
-        -- summary can be used to check all of the blocks in the fragment
-        return $
-          checkFragment
-            (hardForkSummary cfg (VF.validatedLedger validated))
-            now
-            (VF.validatedFragment validated)
+      checkInFuture = \ledgerState af -> do
```

Comment:
Since `checkFragment` has an `error` case in it, please add a comment here explaining how the assumptions made by `checkInFuture` guard this call to `checkFragment`.